### ISSUE TITLE
Prettier needs to be run twice to format {" "} with a space after it

### ISSUE THREAD

**Issue by aaronjensen** [prettier/prettier#2680]
```js
const Foo = () => (
  <div>
    {completionEnabled &&
    notifyNoDoneLists && (
      <span>
        {" "} <Badge src={notificationIconPng} />
      </span>
    )}
  </div>
);
```

formats to:

```js
const Foo = () => (
  <div>
    {completionEnabled &&
    notifyNoDoneLists && (
      <span>
        {" "}
         <Badge src={notificationIconPng} />
      </span>
    )}
  </div>
);
```

and then when run again to:

```js
const Foo = () => (
  <div>
    {completionEnabled &&
    notifyNoDoneLists && (
      <span>
        {" "}
        <Badge src={notificationIconPng} />
      </span>
    )}
  </div>
);
```

Note the space before `<Badge...`

https://prettier.io/playground/#%7B%22content%22%3A%22const%20Foo%20%3D%20()%20%3D%3E%20(%5Cn%20%20%3Cdiv%3E%5Cn%20%20%20%20%7BcompletionEnabled%20%26%26%5Cn%20%20%20%20notifyNoDoneLists%20%26%26%20(%5Cn%20%20%20%20%20%20%3Cspan%3E%5Cn%20%20%20%20%20%20%20%20%7B%5C%22%20%5C%22%7D%20%3CBadge%20src%3D%7BnotificationIconPng%7D%20%2F%3E%5Cn%20%20%20%20%20%20%3C%2Fspan%3E%5Cn%20%20%20%20)%7D%5Cn%20%20%3C%2Fdiv%3E%5Cn)%3B%5Cn%5Cnconst%20Foo%20%3D%20()%20%3D%3E%20(%5Cn%20%20%3Cdiv%3E%5Cn%20%20%20%20%7BcompletionEnabled%20%26%26%5Cn%20%20%20%20notifyNoDoneLists%20%26%26%20(%5Cn%20%20%20%20%20%20%3Cspan%3E%5Cn%20%20%20%20%20%20%20%20%7B%5C%22%20%5C%22%7D%5Cn%20%20%20%20%20%20%20%20%20%3CBadge%20src%3D%7BnotificationIconPng%7D%20%2F%3E%5Cn%20%20%20%20%20%20%3C%2Fspan%3E%5Cn%20%20%20%20)%7D%5Cn%20%20%3C%2Fdiv%3E%5Cn)%3B%22%2C%22options%22%3A%7B%22printWidth%22%3A80%2C%22tabWidth%22%3A2%2C%22singleQuote%22%3Afalse%2C%22trailingComma%22%3A%22none%22%2C%22bracketSpacing%22%3Atrue%2C%22jsxBracketSameLine%22%3Afalse%2C%22parser%22%3A%22babylon%22%2C%22semi%22%3Atrue%2C%22useTabs%22%3Afalse%2C%22doc%22%3Afalse%2C%22ast%22%3Afalse%7D%7D

**Comment by azz** [comment]
cc @karl

**Comment by vjeux** [comment]
I spotted this on the facebook codebase and a few dozens of files are hitting this. It would be nice to get it fixed but it's not the end of the world if we can't fix it, we can just run prettier twice and it's going to be stable from now on and you're unlikely going to write this code often.

**Comment by karl** [comment]
Thanks for the heads up!

I'd like to take a look at fixing this, I'm hoping it it shouldn't take long. It might be a few weeks until I have the spare time to investigate though. Happy for someone else to take a look in the mean time if it becomes more urgent.

**Comment by duailibe** [comment]
I had some free time to investigate. The problem seems to happen everytime a JSXText ends/starts with whitespace followed/preceded by a `{" "}`. Example: `<div>a {" "} b</div>` or even `<div> {" "} </div>`.

In `printJSXElement`, `{" "}` is converted to a JSXText, so when `printJSXChild` is called the children are something like:
```js
[
  {
    type: "JSXText",
    value: "a ",
    raw: "a ",
  },
  {
    type: "JSXText",
    value: " ",
    raw: " ",
  },
  {
    type: "JSXText",
    value: " b",
    raw: " b",
  }
]
```
I'm not sure of what should be the correct fix though.. if the `{" "} -> JSXText` conversion should take the preceding/following child in consideration, or when printing in `printJSXChild`...

I could tackle this if someone could point the right direction here.

**Comment by karl** [comment]
☝🏻 I have a fix for this issue ready to go!

**Comment by azz** [comment]
Fixed in #2973
